### ISSUE TITLE
bad interpreter error when creating virtualenv's in directories with '#' in name

### ISSUE THREAD

**Issue by karishay** [pypa/virtualenv#973]
This had me stumped 🐢 

Ran into an issue where I stopped being able to install packages _at all_ so I nuked my install of python, pip and virtualenv and updated all the things. Finally got things to an arguably working state again and discovered that the issue had likely been the name of my directory all along. 💩  

Here's my current system info where I'm able to see this issue:
> **OS:** macSierra 10.12.1
> **Python:** 2.7
> **Virtualenv:** 15.0.3
> **zsh:** 5.2

Here's the steps to replicate it:
> ✅ make directory with a `#` in it
> ✅ create a virtual env inside that directory
> ✅ activate the virtualenv 
> ✅ install a package with pip
> ✅ try to use the pacakge
> ✅ have an angry nap 😠 😴 

After trying to install a package with the virtualenv activated, it lets me know I have a `bad interpreter` and can't find the env version of python (which I can actually weirdly run directly), and instead of stopping the install it appears to install the package globally.

<img width="625" alt="screen shot 2016-11-04 at 1 29 59 pm" src="https://cloud.githubusercontent.com/assets/4828352/20021630/f6f3b6ae-a293-11e6-82ae-905a4a166d7a.png">

Here's me trying to use the package I've installed, but alas, the package does not smile upon me. 

<img width="582" alt="screen shot 2016-11-04 at 1 30 08 pm" src="https://cloud.githubusercontent.com/assets/4828352/20021624/f10dd314-a293-11e6-8a20-3f30419ffe91.png">

Here's you can see python in the virtualenv and flask installed globally. :flushed:

<img width="444" alt="screen shot 2016-11-04 at 1 57 02 pm" src="https://cloud.githubusercontent.com/assets/4828352/20022170/97cbdb7c-a296-11e6-877a-779587330029.png">

**TLDR;** the moral of the story is to never use `#` in directory names! :tada:

**Comment by yan12125** [comment]
Works fine on Arch Linux, kernel 4.8.6. Seems you're using Mac? I guess that's a XNU/Darwin bug.
```
--2016/11/05 (Saturday) 16:59:12-- Up 13d00:08:34 Last update 00:10:18
yen@NEWSLAB-PC12574 ~/
$ mkdir \#issue117

--2016/11/05 (Saturday) 16:59:26-- Up 13d00:08:48 Last update 00:10:32
yen@NEWSLAB-PC12574 ~/
$ cd \#issue117 

--2016/11/05 (Saturday) 16:59:29-- Up 13d00:08:50 Last update 00:10:34
yen@NEWSLAB-PC12574 ~/#issue117/
$ virtualenv venv
Using base prefix '/usr'
New python executable in /home/yen/#issue117/venv/bin/python3
Also creating executable in /home/yen/#issue117/venv/bin/python
Installing setuptools, pip, wheel...done.

--2016/11/05 (Saturday) 16:59:42-- Up 13d00:09:04 Last update 00:10:48
yen@NEWSLAB-PC12574 ~/#issue117/
$ source ./venv/bin/activate
(venv) 
--2016/11/05 (Saturday) 16:59:58-- Up 13d00:09:20 Last update 00:11:04
yen@NEWSLAB-PC12574 ~/#issue117/
$ python
Python 3.5.2 (default, Jun 28 2016, 08:46:01) 
[GCC 6.1.1 20160602] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>
```

**Comment by yan12125** [comment]
@karishay Could you show files in ```issue#117/env/bin```?
```
ls -l "issue#117/env/bin"
```

**Comment by karishay** [comment]
```bash
total 136
-rw-r--r--  1 sburns  staff   2102 Nov  4 20:25 activate
-rw-r--r--  1 sburns  staff   1044 Nov  4 20:25 activate.csh
-rw-r--r--  1 sburns  staff   2198 Nov  4 20:25 activate.fish
-rw-r--r--  1 sburns  staff   1137 Nov  4 20:25 activate_this.py
-rwxr-xr-x  1 sburns  staff    274 Nov  4 20:25 easy_install
-rwxr-xr-x  1 sburns  staff    274 Nov  4 20:25 easy_install-2.7
-rwxr-xr-x  1 sburns  staff    246 Nov  4 20:25 pip
-rwxr-xr-x  1 sburns  staff    246 Nov  4 20:25 pip2
-rwxr-xr-x  1 sburns  staff    246 Nov  4 20:25 pip2.7
lrwxr-xr-x  1 sburns  staff      9 Nov  4 20:25 python -> python2.7
-rwxr-xr-x  1 sburns  staff   2361 Nov  4 20:25 python-config
lrwxr-xr-x  1 sburns  staff      9 Nov  4 20:25 python2 -> python2.7
-rwxr-xr-x  1 sburns  staff  12568 Nov  4 20:25 python2.7
-rwxr-xr-x  1 sburns  staff    253 Nov  4 20:25 wheel
```

**Comment by yan12125** [comment]
Well, I got the same issue on Sierra, too. Unfortunately that's a Mac's kernel bug [1] - it can not handle paths with #. Either stay away from # or stay away from Mac.

[1] https://opensource.apple.com/source/xnu/xnu-3789.21.4/bsd/kern/kern_exec.c.auto.html, line 511
